frontend/src/services/authentication_handler_service/authentication_handler_service.ts

```diff
@@ -0,0 +1,71 @@
+import { BackendAuthenticationInterface } from "../backend_service_interface/backend_service_interface";
+import { CLIENT_ID } from "./authentication_constants";
+
+/** Error that occurs if sign in fails. */
+export class FailureToSignIn extends Error {
+  constructor() {
+    super("Sign in has failed");
+
+    // Explicitly set prototype to allow expect().toThrow() testing.
```

Comment:
Did you actually hit this problem? Or just being cautious? Newer versions of TypeScript shouldn't have this problem, although some older testing harnesses (like maybe Jest?) still run into it

Reply:
I ran into this error for about 20 minutes haha, definitely a Jest thing here.